### PR TITLE
Add GHA build workflows

### PR DESCRIPTION
**Pull request by halamix2**
**Description**

Changes proposed in this pull request:

- Add GHA build workflows

**Related issue(s)**

- #456

### CODE REVIEW

File: .github/workflows/keda-build-pull.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: keda build (pull)
+
+on:
+  pull_request_target:
+    types: [opened, edited, synchronize, reopened, ready_for_review]
```
Comment (anoipm) [inline .github/workflows/keda-build-pull.yaml:5]: Do we need build for docs?
Now I see difference between pull and push build:
```
    paths-ignore:
      - "docs/**"
      - "examples/**"
```

**Review by Cortey — DISMISSED** [review]

File: .github/workflows/keda-build-pull.yaml (hunk 1 above)
Comment (anoipm) [reply .github/workflows/keda-build-pull.yaml:5]: Also in test-infra we have pre-keda-manager-operator-build job with skip_if_only_changed_documentation.

**Review by anoipm — APPROVED** [review]

**Pull request merged**